packages/strapi-helper-plugin/lib/src/components/InputFile/index.js

```diff
@@ -0,0 +1,127 @@
+/**
+ *
+ *
+ * InputFile
+ *
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { cloneDeep, isArray, isObject, isEmpty, last } from 'lodash';
+import cn from 'classnames';
+
+import ImgPreview from 'components/ImgPreview';
+import InputFileDetails from 'components/InputFileDetails';
+
+import styles from './styles.scss';
+
+class InputFile extends React.Component {
+  state = {
+    isOpen: false,
+    isUploading: false,
+    position: 0,
+  };
+
+  addFilesToProps = (files) => {
+    const initAcc = this.props.multiple ? cloneDeep(this.props.value) : [];
+    const value = Object.keys(files).reduce((acc, current) => {
```

Comment:
An unnecessary loop I think...

```js
addFilesToProps = (files) => {
  const initialValue = this.props.multiple ? cloneDeep(this.props.value) : [];
  const value = initialValue.concat(this.props.multiple ? files : files[0] || []);
```

Reply:
I receive an object here and I need to update my reducer according to the pictures received. If the user uploads multiple files I receive something like this { 0: FIleList, 1: FileList, ... }